reproman/resource/ssh.py

```diff
@@ -180,16 +180,11 @@ def _execute_command(self, command, env=None, cwd=None, handle_permission_denied
         # TODO -- command_env is not used etc...
         # command_env = self.get_updated_env(env)
         from invoke.exceptions import UnexpectedExit
-        command = command_as_string(command)
-        if env:
-            command = ' '.join(['%s=%s' % k for k in env.items()]) \
-                      + ' ' + command
-
-        if cwd:
-            raise NotImplementedError("implement cwd support")
+        command = self._prefix_command(command, env=env, cwd=cwd,
+            with_shell=False)
 
         try:
-            result = self.connection.run(command, hide=True)
+            result = self.connection.run(command_as_string(command), hide=True)
```

Comment:
If _prefix_command were to always return a string, the above command_as_string call could be dropped.  (I think we also need to worry about double-quoting the string.)